pkg/client/dtclient/client_settings.go

```diff
@@ -0,0 +1,307 @@
+		Coordinate coordinate.Coordinate
+		// SchemaId is the Dynatrace settings schema ID
+		SchemaId,
+		// SchemaVersion is the version of the schema
+		SchemaVersion,
+		// Scope is the scope of the schema
+		Scope string
+		// Content is the rendered config for the given settings object
+		Content []byte
+		// OriginObjectId is the object id of the Settings object when it was downloaded from an environment
+		OriginObjectId string
+	}
+
+	SchemaList []struct {
+		SchemaId string `json:"schemaId"`
+	}
+
+	// SchemaListResponse is the response type returned by the ListSchemas operation
+	SchemaListResponse struct {
+		Items      SchemaList `json:"items"`
+		TotalCount int        `json:"totalCount"`
+	}
+
+	// Schema represents the definition of a specific settings schema
+	Schema struct {
+		ID               string
+		UniqueProperties [][]string
+	}
+
+	postResponse struct {
+		ObjectId string `json:"objectId"`
+	}
+
+	settingsRequest struct {
+		SchemaId      string `json:"schemaId"`
+		ExternalId    string `json:"externalId,omitempty"`
+		Scope         string `json:"scope"`
+		Value         any    `json:"value"`
+		SchemaVersion string `json:"schemaVersion,omitempty"`
+		ObjectId      string `json:"objectId,omitempty"`
+	}
+
+	// schemaDetailsResponse is the response type returned by the fetchSchemas operation
+	schemaDetailsResponse struct {
+		SchemaId          string `json:"schemaId"`
+		SchemaConstraints []struct {
+			Type             string   `json:"type"`
+			UniqueProperties []string `json:"uniqueProperties"`
+		} `json:"schemaConstraints"`
+	}
+)
+
+func (d *DynatraceClient) ListSchemas() (schemas SchemaList, err error) {
+	d.limiter.ExecuteBlocking(func() {
+		schemas, err = d.listSchemas(context.TODO())
+	})
+	return
+}
+
+func (d *DynatraceClient) listSchemas(ctx context.Context) (SchemaList, error) {
+	u, err := url.Parse(d.environmentURL + d.settingsSchemaAPIPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url: %w", err)
+	}
+
+	// getting all schemas does not have pagination
+	resp, err := rest.Get(ctx, d.client, u.String())
+	if err != nil {
+		return nil, fmt.Errorf("failed to GET schemas: %w", err)
+	}
+
+	if !resp.IsSuccess() {
+		return nil, rest.NewRespErr(fmt.Sprintf("request failed with HTTP (%d).\n\tResponse content: %s", resp.StatusCode, string(resp.Body)), resp).WithRequestInfo(http.MethodGet, u.String())
+	}
+
+	var result SchemaListResponse
+	err = json.Unmarshal(resp.Body, &result)
+	if err != nil {
+		return nil, rest.NewRespErr("failed to unmarshal response", resp).WithRequestInfo(http.MethodGet, u.String()).WithErr(err)
+	}
+
+	if result.TotalCount != len(result.Items) {
+		log.Warn("Total count of settings 2.0 schemas (=%d) does not match with count of actually downloaded settings 2.0 schemas (=%d)", result.TotalCount, len(result.Items))
+	}
+
+	return result.Items, nil
+}
+
+func (d *DynatraceClient) FetchSchemas(schemaID string) (schema Schema, err error) {
```

Comment:
```suggestion
func (d *DynatraceClient) GetSchema(id string) (schema Schema, err error) {
```

Also one general note - I know we are not doing it in many places but "in general" it is more idiomatic to return `*Schema` instead of `Schema` as the second approach may make it harder to distinguish between a valid schema and an error condition that results in no schema being found.